### PR TITLE
Improve Weblogic patching routine, Fix Process Scheduler feature validation, Remove DeepRemove2 dependency

### PR DESCRIPTION
**Pull request by umaritimus**
* Eliminate DeepRemove2 dependency
* Fully-qualify java executable to fix path errors

### CODE REVIEW

**Comment by iversond** [comment]
Would you want to use `robocopy` for the deep remove? Is that part of Windows Core?

**Comment by umaritimus** [comment]
robocopy definitely works on Windows Core.  It takes 13 seconds on my server to delete weblogic patch directory, so it's definitely not a significant overhead... 

> 
> 
> Would you want to use `robocopy` for the deep remove? Is that part of Windows Core?

**Comment by umaritimus** [comment]
Also adding `prcs_features` validation, please let me know if you need a separate PR

**Comment by bloodmc** [comment]
Looks good. Before I pull, please remove DeepRemove2 from the README as it is no longer required.

**Comment by umaritimus** [comment]
`DeepRemove2` is still used for your custom tuxedo patch deployment.  Would you like me to push my tuxedo patching patch that pretty much undoes your patch?

File: patches/8.56.10/pt_dpk_85610.patch [hunk 1]
```diff
@@ -447,53 +460,16 @@ index 693f506..db1b021 100644
...
--          system(patch_apply_cmd)
--          if $? == 0
--            Puppet.debug("Tuxedo Patch #{tuxedo_patch_num} install successful")
--          else
--            Puppet.debug("Tuxedo Patch #{tuxedo_patch_num} install failed")
-+          tuxedo_patch_dir = tuxedo_patch_dir.gsub('/', '\\')
-+
-+          patch_apply_cmd = "cd #{tuxedo_patch_dir} && #{tuxedo_opatch_cmd} apply #{tuxedo_patch_num} -silent"
-+
-+          begin
```
Comment (bloodmc) [inline patches/8.56.10/pt_dpk_85610.patch]: The begin/ensure/rescue section was added to catch errors in situations where a patch fails. Just replace the DeepRemove2 calls with robocopy. 

Link to ruby exception documentation can be found here
https://ruby-doc.org/core-2.2.0/doc/syntax/exceptions_rdoc.html

File: patches/8.56.10/pt_dpk_85610.patch [hunk 2]
```diff
@@ -447,53 +460,16 @@ index 693f506..db1b021 100644
...
-+              # Fallback to standard removal
-+              # Note: This will fail to remove deep nested folders
-+              remove_cmd = "rmdir /Q /S #{tuxedo_patch_dir}"
-+              system(remove_cmd)
-+            end
-           end
-           # start the services
-           Puppet.debug("Starting Tuxedo ProcManager service [#{tuxedo_procmgr_service}]")
-@@ -424,6 +442,7 @@ Puppet::Type.type(:pt_deploy_tuxedo).provide :deploy_tuxedo,
++          patch_apply_cmd = "cd #{tuxedo_patch_dir} && #{tuxedo_opatch_cmd} apply #{tuxedo_patch_num} -silent < #{patch_response_file}"
```
Comment (bloodmc) [inline patches/8.56.10/pt_dpk_85610.patch]: I removed `patch_response_file` so this will not work.

**Review by bloodmc — APPROVED** [review]
New changes look good.

**Pull request merged**